components/EmotionForm.js

```diff
@@ -84,15 +48,16 @@ export default function EmotionForm({ onCreateEmotion }) {
 
   return (
     <>
-      <StyledSubheadline>Add new Emotion</StyledSubheadline>
+      <StyledSubheadline>
+        {defaultValue ? "Update your Emotion:" : "Add your Emotion:"}
+      </StyledSubheadline>
       <StyledEmotionForm onSubmit={handleSubmit}>
         {/* Dynamic label for emotion, red only after form submission if no emotion is selected */}
-        <StyledLabel htmlFor="emotion" $hasError={hasError}>
+        <label htmlFor="emotion" $hasError={hasError}>
           Emotion (type)*
-        </StyledLabel>
+        </label>
         <select
-          value={selectedEmotion}
-          onChange={handleEmotionChange}
+          defaultValue={defaultValue?.emotion || ""}
```

Comment:
This is not new in this PR, but I noticed that you are using the `index` as the `key` in `emotions.map`. We should not use the `index` because it changes, but rather use an `id`. You will probably still need `id`s anyway. So it might be smart to adapt the data model accordingly and instead of a simple array (your `emotions.js`) use an array with objects instead, i.e. one object per emotion, consisting of name and id. You could then reference this here.

Reply:
Awesome hint! We refactored it to id!